src/index.ts

```diff
@@ -1,9 +1,31 @@
+import {
```

Comment:
Can we just use `export Foo from 'foo'` syntax here? Or is there some sort of optimization thing I'm forgetting about?

Reply:
yes, I re-organized to put type exports at the top, take a peek. Not sure if we can continue to use `export * from...` once the types are in the files, we shall see